### PR TITLE
#29 create accounts for new user

### LINKED ISSUE #29: Server: Create Nextcloud and mail accounts when creating new user
* When creating a new member the corresponding nextcloud and mail accounts should be created accordingly

### PR DESCRIPTION
**Pull request by LennartRukower**
Added nextcloud, mediawiki and mail account creation (currently commented out -> This is going to be resolved before the first deployment).
Added check if name of the user already exists in the database (then a user with v.nachname1 is going to be created).

How to test:
1. Use following POST method (after logging in):
`https://localhost:3030/users` with body
"{
    "vorname": "lennart",
    "nachname":"rukower",
    "name":"l.rukower",
    "geschlecht":"1",
    "password":"kgnjh7i5h4teg",
    "generation":"14",
    "traineeSeit":"2012-12-01",
    "email":"le.rukower@gmail.com",
    "geburtsdatum":"1990-06-06",
    "handy":"49079548954"
}" with an API client like thunder client for example
2. Call this route mutliple times and check in the databse if multiple instances of l.rukower were created (l.rukower, l.rukower1, l.rukower2 etc.)

Closes #29

### CODE REVIEW

File: server/src/members/membersController.ts [hunk 1]
```diff
@@ -275,37 +313,317 @@ export const retrieveCurrentDirectors = (req: Request, res: Response): void => {
...
+  if (req.body.name) {
+    // Search fot req.body.name to check if it already exists
+    const resultFirstQuery = await database.query(`SELECT name FROM mitglied WHERE name = ?`, [req.body.name]);
+    // Check if req.body.name already exists
+    if (Array.isArray(resultFirstQuery) && resultFirstQuery.length === 0) {
+      newUserName = req.body.name;
+    }
+    // Counter for the number of "duplicates" in the database
+    let duplicateCounter = 1;
+    // If name already is taken than create name v.nachname1 (or v.nachname2 etc.)
```
Comment (jbt-it) [inline server/src/members/membersController.ts:332]: Grammar like yoda

File: server/src/utils/nextcloud.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+/**
+ * The default quota of a new user
+ */
+const DEFAULT_QUOTA = "2GB";
+
+/**
+ * The config of the nextcloud api connection
+ */
+const connectionConfig = {
+  baseURL: `http://${NC_HOST}:${NC_PORT}/ocs/v1.php/cloud`,
```
Comment (jbt-it) [inline server/src/utils/nextcloud.ts:37]: http or https?

**Review by jbt-it — APPROVED** [review]
Return the username, ID and e-mail after creating a new user.

File: server/src/utils/nextcloud.ts (hunk 2 above)
Comment (LennartRukower) [reply server/src/utils/nextcloud.ts:37]: Needs to be checked before deployment.
Currently to use it with docker http is necessary.

**Pull request merged**